source/OrderedTypes/SupLattice.lagda

```diff
@@ -0,0 +1,303 @@
+
+partial-orderedness-of : (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                       → is-partial-order ⟨ L ⟩ (order-of L)
+partial-orderedness-of (A , (_≤_ , ⋁_) , order , is-lub-of) = order
+
+reflexivity-of : (L : Sup-Lattice 𝓤 𝓦 𝓥) → is-reflexive (order-of L) holds
+reflexivity-of L = pr₁ (pr₁ (partial-orderedness-of L))
+
+antisymmetry-of : (L : Sup-Lattice 𝓤 𝓦 𝓥) → is-antisymmetric (order-of L) 
+antisymmetry-of L = pr₂ (partial-orderedness-of L)
+
+transitivity-of : (L : Sup-Lattice 𝓤 𝓦 𝓥) → is-transitive (order-of L) holds
+transitivity-of L = pr₂ (pr₁ (partial-orderedness-of L))
+
+join-is-lub-of : (L : Sup-Lattice 𝓤 𝓦 𝓥)
+               → (U : Fam 𝓥 ⟨ L ⟩)
+               → ((order-of L) Joins.is-lub-of join-of L U) U holds
+join-is-lub-of (A , (_≤_ , ⋁_) , order , suprema) = suprema
+
+join-is-upper-bound-of : (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                       → (U : Fam 𝓥 ⟨ L ⟩)
+                       → ((order-of L) Joins.is-an-upper-bound-of
+                          join-of L U) U holds
+join-is-upper-bound-of L U = pr₁ (join-is-lub-of L U)
+
+join-is-least-upper-bound-of : (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                             → (U : Fam 𝓥 ⟨ L ⟩)
+                             → ((u' , _) : Joins.upper-bound (order-of L) U)
+                             → (order-of L (join-of L U) u') holds
+join-is-least-upper-bound-of L U = pr₂ (join-is-lub-of L U)
+
+sethood-of : (L : Sup-Lattice 𝓤 𝓦 𝓥) → is-set ⟨ L ⟩
+sethood-of L =
+ type-with-prop-valued-refl-antisym-rel-is-set
+  (λ x → λ y → order-of L x y holds)
+  (λ x → λ y → holds-is-prop (order-of L x y))
+  (λ x → reflexivity-of L x)
+  (λ x → λ y → antisymmetry-of L)
+
+\end{code}
+
+We now define monotone endomaps on a sup-lattice and specify monotone endomaps
+as a special case.
+
+\begin{code}
+
+module _ where
+
+ is-monotone : {𝓤 𝓤' 𝓦 𝓦' 𝓥 𝓥' : Universe}
+               {L : Sup-Lattice 𝓤 𝓦 𝓥} {M : Sup-Lattice 𝓤' 𝓦' 𝓥'}
+             → (f : ⟨ L ⟩ → ⟨ M ⟩) → 𝓤 ⊔ 𝓦 ⊔ 𝓦'  ̇
+ is-monotone {𝓤} {𝓤'} {𝓦} {𝓦'} {𝓥} {𝓥'} {L} {M} f = (x y : ⟨ L ⟩)
+                                                  → (x ≤⟨ L ⟩ y) holds
+                                                  → (f x ≤⟨ M ⟩ f y) holds
+
+ is-monotone-endomap : {𝓤 𝓦 𝓥 : Universe} {L : Sup-Lattice 𝓤 𝓦 𝓥}
+                     → (f : ⟨ L ⟩ → ⟨ L ⟩) → 𝓤 ⊔ 𝓦  ̇
+ is-monotone-endomap {𝓤} {𝓦} {𝓥} {L} f =
+   is-monotone {𝓤} {𝓤} {𝓦} {𝓦} {𝓥} {𝓥} {L} {L} f
+
+\end{code}
+
+We now show that when one subset contains another the join of their total
+spaces are ordered as expected. 
+
+\begin{code}
+
+module _
+        {𝓤 𝓦 𝓥 : Universe}
+        (L : Sup-Lattice 𝓤 𝓦 𝓥)
+        {A : 𝓥  ̇}
+        (m : A → ⟨ L ⟩)
+       where
+
+ open Joins (order-of L)
+
+ joins-preserve-containment : {P : 𝓟 {𝓥} A} {Q : 𝓟 {𝓥} A}
+                            → P ⊆ Q
+                            → ((⋁⟨ L ⟩ 【 m , P 】)
+                             ≤⟨ L ⟩ (⋁⟨ L ⟩ 【 m , Q 】)) holds
+ joins-preserve-containment {P} {Q} C =
+   (join-is-least-upper-bound-of L 【 m , P 】)
```

Comment:
Minor: the default indentation in TypeTopology is a single space. Please check/fix everywhere.